### PR TITLE
elastic/store: use wait instead of get for barrier

### PR DESCRIPTION
**Pull request by d4l3k**
Summary: We call `.get` in the elastic store barrier operation but we don't need the result. This switches it to use `.wait` instead which eliminates one network round trip as `get` internally does a wait first.

Test Plan:

CI + existing tests -- no behavior change

Differential Revision: D59396199

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @chauhang

### CODE REVIEW

**Review by kurman — APPROVED** [review]
good catch, stamping to unblock - @d4l3k can you fix the store unit tests mocking before landing?

**Review by wconstab — APPROVED** [review]

**Pull request closed without merging**
